tests/test_metrics.py

```diff
@@ -0,0 +1,72 @@
+from unittest import mock
+
+import jax.numpy as jnp
+import pytest
+from jax import random
+
+from blackjax.inference import metrics
+
+KEY = random.PRNGKey(0)
+DTYPE = "float32"
+
+@pytest.fixture()
+def patch_jax(monkeypatch):
+    """Patches JNP multiply and dot to determine which one is called"""
+    multiply, dot = mock.MagicMock(), mock.MagicMock()
+    monkeypatch.setattr(metrics.jnp, "multiply", multiply)
+    monkeypatch.setattr(metrics.jnp, "dot", dot)
+    return multiply, dot
+
+
+def test_gaussian_euclidean_ndim_1(patch_jax):
+    """Test Gaussian Euclidean Function returns correct function when ndim is one"""
+    multiply, _ = patch_jax
+    x = jnp.ones(shape=1)
+
+    assert multiply.call_count == 0
+    momentum, velocity = metrics.gaussian_euclidean(x)
+
+    try:
+        momentum(KEY, [1])
+    except RecursionError:
+        pass
+    assert multiply.call_count == 1
+
+
+def test_gaussian_euclidean_ndim_2(patch_jax):
+    """Test Gaussian Euclidean Function returns correct function when ndim > 1"""
+    _, dot = patch_jax
+    x = jnp.ones(shape=(2, 2))
+
+    assert dot.call_count == 0
+    momentum, velocity = metrics.gaussian_euclidean(x)
+
+    try:
+        momentum(KEY, [1])
+    except RecursionError:
+        pass
+    assert dot.call_count == 1
+
+
+def test_gaussian_euclidean_ndim_3():
+    """Test Gaussian Euclidean Function returns correct function when ndim > 1"""
+    x = jnp.ones(shape=(1, 2, 3))
+
+    with pytest.raises(ValueError) as e:
+        metrics.gaussian_euclidean(x)
+    assert "The mass matrix has the wrong number of dimensions" in str(e)
+
+
+def test_gaussian_euclidean_dim_1():
```

Comment:
@rlouf This one better? If so I'll test kinetic energy, add an ndim 2 test and make all the other changes too

Reply:
Yes! You can add a few edge cases that you think are important. For instance, when testing distributions in MCX I made sure that for a random variable with a stricly positive support `logpdf(0)` returned `-inf` as this is an important edge case.

Reply:
What would the edge cases here be? Negative mass matrix? Other than that the other calculations look pretty straightforward